### DISCUSSION TITLE
API

### DISCUSSION THREAD

**Discussion by tomghub** [haumacher/phoneblock#86] [Ideas]
Hi
wie bekannt, kann die Fritzbox ja nur ca. 4000 Einträge ohne Probleme speichern. Es gibt aber schon 2-3x soviele Einträge. Könnte man da nicht AVM fragen ob die die API einbinden könnten, damit bei jedem Anruf ein Lookup gemacht wird?
Und wenn man jetzt z.b. nur 2000 Einträge synchronisiert, werden dann die aktuellsten Einträge verwendet oder nur ein fester Anteil?

**Comment by haumacher** [comment]
So eine API direkt in der Fritz!Box wäre natürlich die umweltfreundlichste Lösung, da sie am wenigsten Datenverkehr produzieren würde. Aus Privacy-Gesichtspunkten ist so eine Funktion natürlich problematisch, da man damit PhoneBlock über jeden seiner Anrufe informieren würde. Klar, man könnte Anrufer, die sowieso schon im eigenen Adressbuch stehen ausnehmen - also das nur für neue Nummern machen, aber trotzdem habe nur wenig Hoffnung, dass AVM auf solche Vorschläge eingehen wird.

Die Größenbeschränkung kann man aber auch mit dem [PhoneBlock-Anrufbeantworter](
https://phoneblock.net/phoneblock/anrufbeantworter/) umgenen (jetzt ganz neu). 
Der Anrufbeantworter wird als VOIP-Telefon an die FRITZ!Box angemeldet und geht immer dann ran, wenn ein Anruf von einer Nummer auf der Blockliste ankommt. Das Telefon bleibt dann stumm, oder klingelt nur einmal kurz.

Ausprobieren: https://phoneblock.net/phoneblock/anrufbeantworter/

**Reply by tomghub** [reply]
Das ist ja absolut genial. hoffentlich verursacht diese Funktion nicht zuviel Last auf dem Server durch die SIP Anmeldung und die Gespräche.
Ich habe es getestet und da werden tatsächlich auch Ansagen gespielt wie z.b. hier erwähnt: https://www.linux-magazin.de/ausgaben/2008/04/gemeinschaft-an-der-strippe/4/
Jetzt fehlt nur noch die Funktion das man sich die Aufnahmen anhören könnte aber das verursacht ja auch nur Speicherplatz.
